### PR TITLE
Directly: Customize styles

### PR DESCRIPTION
**Pull request by mattwondra**
Creates customized styles for Directly's Real Time Messaging widget. The goal is to make the RTM widget feel more native to Calypso and more similar to the Happychat interface.

Due to the widget's architecture, a few extra steps are needed to view our custom CSS in development. **Please read the comment block at the top of [assets/stylesheets/directly.scss](https://github.com/Automattic/wp-calypso/pull/11600/files#diff-ace0466620b3191b23dec85ee11ecc08) for instructions to view the custom CSS in development.**

### To view this in development
* Restart your local Calypso (to build the new CSS file)
* Install the Chrome extension [Switcharoo Redirector](https://chrome.google.com/webstore/detail/switcheroo-redirector/cnmciclhnghalnpfhhleggldniplelbg) and use it to redirect the URL `https://wpcalypso.wordpress.com/calypso/directly.css` to `http://calypso.localhost:3000/calypso/directly.css`
* Quit Chrome and re-open, temporarily allowing insecure content. On OS X open Terminal and issue: `open -a Google\ Chrome --args --allow-running-insecure-content` (**make sure you quit and re-open Chrome normally once done testing**)
* Sign in as a user with no paid upgrades and head to calypso.localhost:3000/help/contact
* You should see a form with a button "Ask an Expert". Enter a question and press this button and the Directly RTM widget will appear.

### Feedback requested
- This PR adds a brand new CSS bundle. It seems to be building fine in development, but did we miss any steps/config that would cause problems in stage/production?
- Does the documentation in [assets/stylesheets/directly.scss](https://github.com/Automattic/wp-calypso/pull/11600/files#diff-ace0466620b3191b23dec85ee11ecc08) clearly explain steps to view & adjust CSS in development?
- General CSS review.

#### Before

screen shot 2017-03-02 at 7 47 31 am

#### After

screen shot 2017-03-02 at 7 46 02 am

### CODE REVIEW

**Comment by mattwondra** [comment]
@mtias Do you have a few minutes to gut-check the approach here for creating a new production CSS bundle? (If not or there's someone else you think would be better to look, feel free to pass this on to them.)

To support the Directly third-party widget integration we need to create a new CSS file at `public/editor.css`. The widget loads an iframe, and this CSS file will be `<link>`'d from inside the iframe. You can read more in the comment block at the top of [assets/stylesheets/directly.scss](https://github.com/Automattic/wp-calypso/pull/11600/files#diff-ace0466620b3191b23dec85ee11ecc08)

To create `public/directly.css` I followed the patterns I saw for `public/editor.css` and the like, and it seems to be working perfectly in development. Is what I've done the right approach, and is there anything else I need to do to ensure this will behave properly in staging/production environments?

File: assets/stylesheets/directly.scss [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+	cursor: pointer;
+	font-size: 14px;
+	font-weight: 500;
+	vertical-align: top;
+	box-sizing: border-box;
+	line-height: 21px;
+	border-radius: 4px;
+	padding: 7px 14px 9px;
+	-webkit-appearance: none;
+	-moz-appearance: none;
```
Comment (omarjackman) [inline assets/stylesheets/directly.scss]: We're using an autoprefixer so I think we don't need to explicitly put `-moz` and `-webkit` here.

File: assets/stylesheets/directly.scss [hunk 2]
```diff
@@ -0,0 +1,503 @@
...
+.maximized-view {
+	background: #F2F6F8;
+}
+.minimized-view {
+	background: $white;
+	box-shadow: 0 1px 10px rgba(46,68,83,.1);
+	border: 1px solid lighten( $gray, 25% );
+}
+.maximized-view.is-ask-view {
+	top: 0 !important;
```
Comment (omarjackman) [inline assets/stylesheets/directly.scss]: Maybe `.chat-container .maximized-view.is-ask-view`

File: assets/stylesheets/directly.scss [hunk 3]
```diff
@@ -0,0 +1,503 @@
...
+			position: absolute;
+			content: "";
+			display: block;
+			width: 0;
+			height: 0;
+			border-style: solid;
+			border-width: 10px 10px 0 0;
+			border-color: #14abdd transparent transparent transparent;
+			top: 0;
+			right: -10px;
```
Comment (mattwondra) [inline assets/stylesheets/directly.scss]: Yep you're right! https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/css.md#positioning

I'll make that change for the places where `position` is explicitly used.

**Comment by melchoyce** [comment]
Do we have any control over those icons? Would be good to swap them out for Gridicons, if possible.

**Comment by folletto** [comment]
From the screenshot it seems good enough to me to be pushed in, and we can review later on the rest. It's a third party plugin so I expect we have constraints.

As far as I understand, given:

1. The generated file is separate and doesn't impact Calypso except a couple of lines (`directly-rtm`).
2. There's no style leakage as it's an iframe.
3. The CSS is loaded separately from their admin interface.

We can ship this and polish incrementally later.

**Comment by mattwondra** [comment]
> Do we have any control over those icons? Would be good to swap them out for Gridicons, if possible.

@melchoyce I haven't used Gridicons but we don't have any control over the markup here and it looks like you need to use React to use the icons. If I'm wrong and there's a supported way to use them with just CSS & static images, we could theoretically use them.

> We can ship this and polish incrementally later.

@folletto Is that an official 👍 or should I wait for an in-depth review from someone like @jasmussen?

**Comment by folletto** [comment]
If it can wait tomorrow, to see if he has a moment, then let's wait.
If it's better to push it live, we can push it live.

:)

**Comment by melchoyce** [comment]
> I haven't used Gridicons but we don't have any control over the markup here and it looks like you need to use React to use the icons. If I'm wrong and there's a supported way to use them with just CSS & static images, we could theoretically use them.

Yup, you can use [the SVGs](https://github.com/Automattic/gridicons/tree/master/svg) in CSS background-images: https://css-tricks.com/using-svg/#article-header-id-4

**Comment by jasmussen** [comment]
Nice work!

Yeah using Gridicons outside of React right now is a bit of a pain. Though @keoshi recently made it easier, by letting you reference an external sprite and then inserting the icons using shorter command. That's what we do on http://automattic.github.io/gridicons/. 

However that would still require you editing the markup. And though you can use SVGs as backgrounds, that is a ton of work if there are many icons in many colors, and not very maintainable. I would put "replace icons with Gridicons" as something that'll earn you a 🏅 , but not a blocker for merge (though if we don't replace them with gridicons, probably worth opening a separate ticket to do so in a second iteration). 

You could also carefully consider which icons are absolutely needed. Some seem decorative to the point that we can probably do without them. I could see the "testimonial sent" icon disappearing, perhaps even the checkmark in accepted answer. Though the "Send" icon would be nice to replace.

Just to be sure, the "Type your reply here" box is disabled right? Otherwise it should be white. 

Can the entire sidebar be a wee bit thinner? A la Happychat width? It covers a lot right now. The avatar could also be made a bit smaller. 

Also that green smiley — not familiar with this particular chat interface, but it kind of stands out. Would be interested in anything we can do here, but assuming not much.

The green color of "accepted answer" should come from https://wordpress.com/design-handbook/colors/. Not sure if it does at the moment. If it does, 👍 

None of those are blockers really, agree with @folletto on that. Just good to create a V2 ticket if any of the above _can_ be fixed later on, so it doesn't fall between the cracks. 

Nice work.

**Comment by mattwondra** [comment]
Thanks for the input everyone. This project is still a trial, so I'll round up these comments for next phases if we stick with the Directly product long-term.

**Pull request merged**